docs/current_docs/manuals/developer/enums.mdx

```diff
@@ -0,0 +1,86 @@
+---
+slug: /manuals/developer/enums
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Enums"
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Enums
+
+
+Dagger supports custom enumeration types, this is useful to restrict possible values for an arguments.
+
+:::note
+Following the [GraphQL specification](https://spec.graphql.org/October2021/#Name), enums are represented as strings in the GraphQL schema and follows the following rules:
+- The value cannot start with digit and can only be composed of letter, digit or _
+- Name are case sensitive, and by convention should be in screaming case
+:::
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+Here is an example of a `trivy` Dagger module with a function named `Scan` that will take as arguments the reference of an image and a severity
+defined as an enum `Severity`:
+
+```go file=./snippets/enums/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+Here is an example of a `trivy` Dagger module with a function named `scan` that will take as arguments the reference of an image and a severity
+defined as an enum `Severity`:
+
+```python file=./snippets/enums/python/main.py
+```
+
+</TabItem>
```

Comment:
I feel that Python needed a little more explanation (e.g., `dagger.Enum` is not actually required) that could be added in a note here, but wrote it in the [reference docs](https://dagger-io--8104.org.readthedocs.build/en/8104/module.html#dagger.enum_type) instead.

Reply:
I see that reference isn't doc isn't published yet, so I'll add a note and we can add the reference later.
Or maybe we can release an update for the Python reference doc before?

Reply:
Added note in [`a6c5321` (#8104)](https://github.com/dagger/dagger/pull/8104/commits/a6c532179099348776ba3f4cd7e383d58a88889a)